src/main/java/uk/gov/register/serialization/CommandParser.java

```diff
@@ -75,17 +80,18 @@ public void addCommand(String s) {
                 break;
             case "append-entry":
                 if (parts.length == 4) {
-                    Entry entry = new Entry(0, stripPrefix(parts[2]), Instant.parse(parts[1]), parts[3]);
+                    Entry entry = new Entry(0, new HashValue(HashingAlgorithm.SHA256, stripPrefix(parts[2])), Instant.parse(parts[1]), parts[3]);
```

Comment:
You should be able to get rid of `stripPrefix` and instead use `decode()` - the benefit of the new `HashValue` object is that all this prefix removal happens in one place (i.e. `decode()`).